pdf/pdfcanvas.cpp

```diff
@@ -350,6 +376,57 @@ QPair<QUrl, PDFCanvas::ReducedBox> PDFCanvas::urlAtPoint(const QPointF &point) c
     return QPair<QUrl, PDFCanvas::ReducedBox>();
 }
 
+QPair<Poppler::Annotation*, PDFCanvas::ReducedBox> PDFCanvas::annotationAtPoint(const QPointF &point) const
+{
+    for (int i = 0; i < d->pageCount; ++i) {
+        const PDFPage &page = d->pages.value(i);
```

Comment:
Yes, I guess a dichotomic algorithm should switch from O(N) to O(ln(N)) since the rectangles of pages are sorted (and almost all of the same size). I mean, int(at.y / height) should give a good guess of page index i0, then trying i0 + int((at.y - i0.y)/(height-i0.height)…